src/transformers/models/llama/modeling_llama.py

```diff
@@ -589,13 +634,13 @@ def _upad_input(self, query_layer, key_layer, value_layer, padding_mask, query_l
 
 
 class LlamaDecoderLayer(nn.Module):
-    def __init__(self, config: LlamaConfig):
+    def __init__(self, config: LlamaConfig, mask_converter=None):
```

Comment:
If we support passing the mask converter here but not in the parent classes it's kind of pointless no?
Wondering which one would be the best:
1. Pass the mask converter class to all classes
2. Only have it in the attention layer, controlled with a `MASK_CONVERTER = {"default": AttentionMask2DTo4D}` and just in the attention layer do `self.mask_converter = MASK_CONVERTER[config.mask_converter]` with the attribute added to the config common? 
(naming can be improve for sure!)

Reply:
Sorry I don't fully understand this

Reply:
To begin with I would not make the Mask Converter a class that the user plays around with, but instead use it as an internal convenience class that doesn't sacrifice speed

Reply:
I see sorry realised that if you want to share the same cached mask you gotta pass it, ignore my comment